### DISCUSSION TITLE
What is the best way to diff records?

### DISCUSSION THREAD

**Discussion by Serpentarius13** [prisma/prisma#22225] [Q&A]
### Question

I have a situation where I need to keep values from frontend in check. For example, I have table User with user_id and table User_Subscribers with user_id subscriber_id
On frontend I have a functionality to add/delete/update users. So, every time I click save, I send a new array which changed unpredictably. What is the best way to diff what needs to be updated, deleted and created in prisma? I've tried just removing all the old records, but that doesn't seem to be a good way to me

**Comment by ludralph** [comment]
Hi @Serpentarius13 👋 

Thank you for raising this question.

One way to accomplish this would be to fetch the current state of the user and their subscribers from the database, compare it with the new array from the frontend, and then determine which records need to be created, updated, or deleted. You would need to use a diffing algorithm to compare the existing data with the new data from the frontend. This will identify the records that need to be created, updated, or deleted.

**Reply by Serpentarius13** [reply]
Hello! Thanks for your answer! Perhaps you know some best-practice approach to this problem? Do I just create a generic function to diff incoming array with the fetched one and use it, or maybe there is something in the prisma that could help me do the diffing?

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
